### DISCUSSION TITLE
Building error on WSServer.cpp(210,2):

### DISCUSSION THREAD

**Discussion by sebionofrei** [Palakis/obs-websocket#827] [Q&A]
Hello guys,

Sorry for disturbing you, but I'm facing the following error, no idea how to... fix it :)

WSServer.cpp(210,2): error C2653: 'QtConcurrent': is not a class or namespace name

Any ideas?

**Comment by tt2468** [comment]
If you are building in-tree with OBS, their version of Qt does not include the QtConcurrent module. You will have to build with a full Qt installation.

**Comment by sebionofrei** [comment]
This is what I do:

cmake "D:/Work/OBS/Qt/5.15.2/msvc2019" -A Win32 -B build_x86 -S . -DCMAKE_INSTALL_PREFIX=dist

Followed by 

cmake --build build_x86 --config Release

**Reply by tt2468** [reply]
Yes, like I said, you need to use a full installation of Qt, not the version that the OBS project provides.

**Comment by sebionofrei** [comment]
That path is the install.
The idea is if I go to D:/Work/OBS/Qt/5.15.2/msvc2019
 --- then i navigate to **\include\QtConcurrent** i can find the QtConcurrent or QtConcurrentRun files in that folder.

But to me it seems that, at compile time, it fails to detect that QtConcurrent is a namespace.

In the WSServer.cpp i modified (for testing) the include headers... for example, modified it to `#include <QtConcurrent/QtConcurrent2>` and compiler throws error it doesn't find the file:

`WSServer.cpp(26,10): fatal error C1083: Cannot open include file: 'QtConcurrent/QtConcurrent2': No such file or directory`

This tells me that the compiler locates the correct include file, but fails to determine that QtConcurrent is a namespace somehow....


If i comment the code about QtConcurrent, it goes forward, but gives other error, at linking:
`WSServer.obj : error LNK2001: unresolved external symbol "public: virtual struct QMetaObject const * __thiscall WSServe
r::metaObject(void)const " (?metaObject@WSServer@@UBEPBUQMetaObject@@XZ)`
